static/app/utils/profiling/flamegraph/flamegraphStateProvider/index.tsx

```diff
@@ -257,19 +277,48 @@ export function FlamegraphStateProvider(
   });
 
   useEffect(() => {
-    if (
-      reducer[0].profiles.threadId === null &&
-      profileGroup.type === 'resolved' &&
-      typeof profileGroup.data.activeProfileIndex === 'number'
-    ) {
-      const threadID =
-        profileGroup.data.profiles[profileGroup.data.activeProfileIndex].threadId;
-
-      if (threadID) {
-        reducer[1]({
-          type: 'set thread id',
-          payload: threadID,
-        });
+    if (reducer[0].profiles.threadId === null && profileGroup.type === 'resolved') {
```

Comment:
Should this instead be done in the reducer init? These state effects are usually tricky to manage as you start to get multiple frames. As an alternative, can we do it when the data is resolved?